pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -1261,4 +1268,23 @@ public static class NullValuePlaceHolder {
     public static final byte[][] BYTES_ARRAY = new byte[0][];
     public static final Object MAP = Collections.emptyMap();
   }
+
+  public static class CursorConfigs {
+    public static final String DEFAULT_RESULT_STORE_TYPE = "file";
+    public static final String DEFAULT_RESULT_SERDE = "json";
+    public static final int MAX_QUERY_RESULT_SIZE = 100000;
+    public static final int DEFAULT_QUERY_RESULT_SIZE = 10000;
+    public static final String DEFAULT_RESULTS_EXPIRATION_INTERVAL = "1h"; // 1 hour.
+    public static final String PREFIX_OF_CONFIG_OF_CURSOR = "pinot.broker.cursor";
+    public static final String PREFIX_OF_CONFIG_OF_RESULT_STORE = "pinot.broker.cursor.response.store";
+    public static final String RESULT_STORE_TYPE = "type";
+    public static final String RESULT_STORE_SERDE = "serde";
+    public static final String QUERY_RESULT_SIZE = PREFIX_OF_CONFIG_OF_CURSOR + ".result.size";
+    public static final String RESULTS_EXPIRATION_INTERVAL = PREFIX_OF_CONFIG_OF_RESULT_STORE + ".expiration";
+
+    public static final String RESULT_STORE_CLEANER_FREQUENCY_PERIOD =
+        "controller.cluster.response.store.cleaner.frequencyPeriod";
+    public static final String RESULT_STORE_CLEANER_INITIAL_DELAY =
+        "controller.cluster.response.store.cleaner.initialDelay";
```

Comment:
Usually defaults are defined close to the property definition. Where are they defined in this case?

Reply:
Default `initialDelay` is random. For `frequencyPeriod` there are a couple of different ways. I have specified the default here.